File: orquestator/helper/twiml.go
package helper

import (
	"os"

	"github.com/twilio/twilio-go/twiml"
)

// End the call with a message
func EndCall(message string) (string, error) {
	twimlResult, err := twiml.Voice([]twiml.Element{
		&twiml.VoiceSay{
			Message:  message,
			Voice:    os.Getenv("TWILIO_VOICE"),
			Language: os.Getenv("TWILIO_LANGUAGE"),
		},
		&twiml.VoiceHangup{},
	})

	return twimlResult, err
}

// Use GatherCall to say something to the client and recive a speech transcription
func GatherCall(message string) (string, error) {
	twimlResult, err := twiml.Voice([]twiml.Element{
		&twiml.VoiceGather{
			Action:          os.Getenv("GATHER_ENDPOINT"),
			Language:        os.Getenv("TWILIO_LANGUAGE"),
			Input:           "speech",
			Enhanced:        "true",
			SpeechTimeout:   "auto",
			ProfanityFilter: "false",
			BargeIn:         "false",
			SpeechModel:     "experimental_conversations",
			InnerElements: []twiml.Element{
				&twiml.VoiceSay{
					Message:  message,
					Voice:    os.Getenv("TWILIO_VOICE"),
					Language: os.Getenv("TWILIO_LANGUAGE"),
				},
			},
		},
	})

	return twimlResult, err
}

File: orquestator/helper/twiml.py
import os
from twilio.twiml.voice_response import VoiceResponse, Gather

TWILIO_VOICE = os.getenv("TWILIO_VOICE", "Polly.Mia")
TWILIO_LANGUAGE = os.getenv("TWILIO_LANGUAGE", "es-MX")
GATHER_ENDPOINT = os.getenv("GATHER_ENDPOINT")

def end_call(message: str) -> str:
    """Termina la llamada con un mensaje"""
    response = VoiceResponse()
    response.say(
        message,
        voice=TWILIO_VOICE,
        language=TWILIO_LANGUAGE
    )
    response.hangup()
    return str(response)

def gather_call(message: str) -> str:
    """Dice algo y espera respuesta por voz"""
    response = VoiceResponse()
    gather = Gather(
        action=GATHER_ENDPOINT,
        input='speech',
        language=TWILIO_LANGUAGE,
        enhanced=True,
        speech_timeout='auto',
        profanity_filter=False,
        barge_in=False,
        speech_model='experimental_conversations'
    )
    gather.say(
        message,
        voice=TWILIO_VOICE,
        language=TWILIO_LANGUAGE
    )
    response.append(gather)
    return str(response)